### ISSUE TITLE
API first development for Nestor-QA

### ISSUE THREAD

**Issue by kinow** [nestor-qa/nestor#81]
Known cases:

* Testopia
* Trello

Premises:

* Security is a concern for this - see Testopia Web Services with Cookies vs. Tokens
* Automatic documentation is good too - like https://django-tastypie.readthedocs.org/en/latest/

**Comment by kinow** [comment]
Security is a concern for this - see Testopia Web Services with Cookies vs. Tokens

**Comment by kinow** [comment]
Automatic documentation is good too - like https://django-tastypie.readthedocs.org/en/latest/

**Comment by kinow** [comment]
Using the teepluss/hmvc and teepluss/restable modules, we are able to use the vanilla Restful controllers in Laravel creating an API, call the methods (via hmvc), and return the elements in a nice JSON object (or xml, etc, thanks to restable). Closing since we've designed a solution. Will propagate this change through the rest of the project as I fix the broken parts. 

The project controller (not in the manage section) is fully API first, including pagination.
